### PR TITLE
Add publisher status report notebook

### PR DESCRIPTION
**Pull request by duncandewhurst**
* Closes https://github.com/open-contracting/notebooks-oc4ids/issues/45

@neelima-j I've documented the outstanding tasks for this PR below. Feel free to move these to issues if you want to get the PR merged. You don't need to do all of these before sharing the report with CoST, but it would be good to complete them before the project is wrapped up.

To do:

**General**
- [x] Add new notebooks to the table in `README.md`
- [x] Format SQL code in new notebooks using [pgFormatter](http://sqlformat.darold.net/) (can reuse the script from https://github.com/open-contracting/notebooks-ocds/blob/624dcca5d1c360132f0f50af7be03daa119b321a/manage.py#L165-L176)
- [x] Ask tech infrastructure group about setting up database backups

**Data import notebook**
- [ ] ~~Refactor so that all imports can be run at once~~ (Moved to https://github.com/open-contracting/notebooks-oc4ids/issues/46)

**Quality criteria, checks and metrics notebook**
- [x] `semantics_coordinates`: Update to cover all supported geometry types
- [ ] ~~`criteria_registered`: Add code to update registered_prefixes table from https://standard.open-contracting.org/staging/infrastructure/0.9-dev/en/reference/prefixes~~ (Moved to https://github.com/open-contracting/notebooks-oc4ids/issues/47)

**Publisher status report notebook**
- [x] Replace `collection_id` in coverage query outputs with `source_id`. See `get_output` function definition for an example.
- [ ] ~~Format coverage query outputs for readability using [Styler](https://pandas.pydata.org/docs/user_guide/style.html), e.g. colour scale for coverage scores, collapsing/expanding for objects. See `get_results` for an example of how to do this.~~ (Moved to https://github.com/open-contracting/notebooks-oc4ids/issues/45)

### CODE REVIEW

**Comment by duncandewhurst** [comment]
@neelima-j please see the list of outstanding tasks in the PR description.

**Comment by neelima-j** [comment]
Outstanding tasks moved to separate issues #45 #46 #47

File: OC4IDS_Publisher_Status_Report.ipynb [hunk 1]
```diff
@@ -0,0 +1,1796 @@
+{
```
Comment (odscjen) [inline OC4IDS_Publisher_Status_Report.ipynb:1]: <p>Using run_id = 2024-08-02 06:29:25.083245</p><p> </p><p>Cell didn't run, got the error</p><p> </p>

<pre class="ql-syntax" spellcheck="false">File "&lt;ipython-input-20-704129bd1c51&gt;", line 5
    source_id,
    ^
IndentationError: unexpected indent
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (odscjen) [inline OC4IDS_Publisher_Status_Report.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    % % sql
</pre><p>should be %%sql i.e. need to remove the spaces</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

Comment (odscjen) [inline OC4IDS_Publisher_Status_Report.ipynb:1]: <p>cell returned error</p><p> </p>

<pre class="ql-syntax" spellcheck="false">KeyError                                  Traceback (most recent call last)
/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py in get_loc(self, key)
   3804         try:
-&gt; 3805             return self._engine.get_loc(casted_key)
   3806         except KeyError as err:

index.pyx in pandas._libs.index.IndexEngine.get_loc()

index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.get_item()

KeyError: 'check'
The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
</pre><p> </p><p> </p><p>6 frames</p><p> </p>

<pre class="ql-syntax" spellcheck="false">/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py in get_loc(self, key)
   3810             ):
   3811                 raise InvalidIndexError(key)
-&gt; 3812             raise KeyError(key) from err
   3813         except TypeError:
   3814             # If we have a listlike key, _check_indexing_error will raise

KeyError: 'check'
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

Comment (odscjen) [inline OC4IDS_Publisher_Status_Report.ipynb:1]: <p>This and the following other date cells, in the returned table the column with the date is titled "count" which is confusing</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

File: OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb [hunk 2]
```diff
@@ -0,0 +1,1792 @@
+{
```
Comment (odscjen) [inline OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb:1]: <p>I don't understand why 2 collections need to be chosen and what the difference should be between them, some explanation is needed, or if there is explanation elsewhere it needs to be signposted here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (odscjen) [inline OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb:1]: <p>This is returning an empty table which means all the other checks also return an empty table, not sure what I'm doing wrong? I used 'dev_1' and 'dev_2' as the default load and comparison_load ids and then also tried with 2 collections I created myself using the import notebook with the same empty result.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/open-contracting/notebooks-oc4ids/pull/43/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by duncandewhurst** [comment]
@neelima-j are you still planning to action the comments on this PR? It would be good to get it merged.

File: OC4IDS_Database_Data_Import.ipynb [hunk 3]
```diff
@@ -16,9 +16,11 @@
         "\n",
```
[the remark this answers is not in the document]
Comment (neelima-j) [reply OC4IDS_Database_Data_Import.ipynb:16]: <p>Fixed by moving all subsequent code to a new heading and clarifying the instructions. </p>

File: OC4IDS_Publisher_Status_Report.ipynb (hunk 1 above)
Comment (neelima-j) [reply OC4IDS_Publisher_Status_Report.ipynb:1]: <p>Fixed the sql magic syntax error</p>

Comment (neelima-j) [reply OC4IDS_Publisher_Status_Report.ipynb:1]: <p>Updated column names to be the same as the name of the check. </p>

File: OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb (hunk 2 above)
Comment (neelima-j) [reply OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb:1]: Explanation added

Comment (neelima-j) [reply OC4IDS_Quality_Criteria_Checks_and_Metrics.ipynb:1]: <p>colab shows this cell as returning an empty table, but the insert command works as expected. </p><p> </p><p>I can see the run_collection table has been populated with the run_id and collections corresponding to dev_1 on the date and time Jen worked on this. </p>

**Comment by neelima-j** [comment]
@duncandewhurst  Resolved @odscjen 's comments. 

In addition to that, I changed the get_results function in the report notebook to fix the following bugs:
-  Summary table was duplicating some manual checks 
-  Newly appended manual checks do not contain explanatory check names, but the check id

**Comment by duncandewhurst** [comment]
@neelima-j I reviewed your changes and they look good, but I can't approve this PR because I created it. Please can you approve and merge it?

Whilst reviewing I made a few commits to fix some issues I noticed.

**Review by neelima-j — APPROVED** [review]

**Pull request merged**